### PR TITLE
optimizer ssl hint for cpp

### LINKED ISSUE #4504: [cpp]not clear log
**Describe the bug**

 i try to code
```
    Client client("pulsar+ssl://**.com:6650");
```

**To Reproduce**
1. Go to 
```  
pulsar-client-cpp/example/SampleProducer.cc
```  

2. change 
```  
Client client("pulsar://localhost:6650"); 
```
  to 
```  
Client client("pulsar+ssl://yoururl:6650");
```  

3. run SampleProducer  

```
libc++abi.dylib: terminating with uncaught exception of type boost::wrapexcept<boost::system::system_error>: load_verify_file: No such file or directory
Abort trap: 6
```

**Expected behavior**
 
print ```No such trustCertFile```

**Screenshots**

**Desktop (please complete the following information):**

**Additional context**

### PR DESCRIPTION
**Pull request by liangyuanpeng**
*(If this PR fixes a github issue, please add `Fixes #<xyz>`.)*

Fixes #4504 

### Motivation

optimize the ssl hint for cpp

*Explain here the context, and why you're making that change. What is the problem you're trying to solve.*

### Modifications

*Describe the modifications you've done.*

### Verifying this change

- [  ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

can change the ``pulsar://`` in the examples/SampleProducer.cc to ``pulsar+ssl://`` for testing

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API: (yes / **no**)
  - The schema: (yes / **no** / don't know)
  - The default values of configurations: (yes / **no**)
  - The wire protocol: (yes / **no**)
  - The rest endpoints: (yes / **no**)
  - The admin cli options: (yes / **no**)
  - Anything that affects deployment: (yes / **no** / don't know)

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

File: pulsar-client-cpp/lib/ClientConnection.cc [hunk 1]
```diff
@@ -119,6 +119,9 @@ static Result getResult(ServerError serverError) {
 }
 
 static bool file_exists(const std::string& path) {
+    if(path.empty()){
```
Comment (merlimat) [inline pulsar-client-cpp/lib/ClientConnection.cc]:
```suggestion
    if (path.empty()){
```

The format checker is failing on the missing space.

**Comment by jiazhai** [comment]
@liangyuanpeng  cpp tests failed for the reason that Matteo pointed out above.  

```
clang-format checks failed, run 'make format' to fix
```

**Comment by sijie** [comment]
@liangyuanpeng @jiazhai there is a script to do so : `./pulsar-client-cpp/docker-format.sh`

**Comment by liangyuanpeng** [comment]
@sijie  @jiazhai @merlimat 
 👍

**Comment by sijie** [comment]
@liangyuanpeng cool

**Comment by sijie** [comment]
run cpp tests

**Comment by sijie** [comment]
run java8 tests

**Pull request merged**